web/src/components/TaxRegistrationModal.tsx

```diff
@@ -56,7 +56,7 @@ export const TaxRegistrationModal = (props: Props): ReactElement => {
         invalid: !profileData.existingEmployees,
       },
       taxId: {
-        invalid: profileData.taxId?.length !== 12 && showTaxIdField(),
+        invalid: profileData.taxId?.length !== 12,
```

Comment:
Given that we're changing this logic, I think the test should change to reflect this new behavior